src/utils/batched-updates.js

```diff
@@ -8,23 +8,33 @@ import {
 import defaults from '../defaults';
 import supports from './supported-features';
 
-let isInsideBatchedSchedule = false;
+let batchedScheduled = false;
+
+let batched = [];
+
+const executeBatched = () => {
+  unstable_batchedUpdates(() => {
+    while (batched.length) {
```

Comment:
Could this be also written as:
```
let fn;
while ((fn = batched.shift())) {
   fn();
}
```

Or am I missing the reason why we would replace batched with an empty array?

Reply:
I have a bad tendency for not using array modification methods as they are "slow".
But this way you can read and write it at the same time without "buffer" being reset.

Simplifies code a lot, 👍